src/Symfony/Component/Lock/Key.php

```diff
@@ -70,4 +71,27 @@ public function getState($stateKey)
     {
         return $this->state[$stateKey];
     }
+
+    public function resetExpiringDate()
+    {
+        $this->expiringDate = null;
+    }
+
+    /**
+     * @param \DateTimeImmutable $expiringDate
+     */
+    public function reduceExpiringDate(\DateTimeImmutable $expiringDate)
+    {
+        if (null === $this->expiringDate || $this->expiringDate > $expiringDate) {
```

Comment:
Why not throw an exception when ``$this->expiringDate <= $expiringDate``?

Reply:
Finally replaced `reduceExpiringDate(\Datetime $newExpiringDate)` by `reduceLifetime(float $ttl)`